File: src/main/java/eotree/data/EORegexData.java
package eotree.data;

/**
 * EBNF representation:
 * <code>
 * / /.+/[a-z]* /
 * </code>
 */
public class EORegexData extends EOData {
    public String regex;

    public EORegexData(String regex) {
        this.regex = regex;
    }

    @Override
    public String generateEO(int indent) {
        return regex;
    }
}

File: src/main/java/eotree/data/EORegexData.kt
package eotree.data

/**
 * EBNF representation:
 * `
 * / /.+/[a-z]* /
` *
 */
class EORegexData(var regex: String) : EOData() {
    override fun generateEO(indent: Int): String {
        return regex
    }
}
